var/spack/repos/builtin/packages/poplddecay/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Poplddecay(MakefilePackage):
+    """
+    PopLDdecay: a fast and effective tool for linkage disequilibrium
+    decay analysis based on variant call format files
+    """
+
+    homepage = "https://github.com/BGI-shenzhen/PopLDdecay"
+    url      = "https://github.com/BGI-shenzhen/PopLDdecay/archive/v3.41.tar.gz"
+
+    maintainers = ['robqiao']
+
+    version('3.41', sha256='09a1ad01581520b84ef73eaa0b199879c84e77b259ba6ff06dbca5fcfc090457')
+    version('3.40', sha256='5070930166fb90f7eaaa4b87c4430caa8a827d79c54683e2f56434a4daf69778')
+
+    def build(self, spec, prefix):
+        ###################
+        # Build Directory #
+        ###################
+        with working_dir('src'):
+            make()
+            make('clean')
+
+    def install(self, spec, prefix):
+        mkdirp(prefix.bin.bin)
+        install('bin/PopLDdecay', prefix.bin.bin)
```

Comment:
What's up with this? I've never seen a package use `prefix.bin.bin` before.

Reply:
Ye, I agree. This step here is to align the install path inline with its documentation, which says as below from `-h` 
OR 
[Manual](https://github.com/BGI-shenzhen/PopLDdecay/blob/master/Manual.pdf)

 Here, we provide four classic cases to demonstrate the application of this software, four situation will be show how to follow to get the LD decay figure out.

1). One population
    This situation (one population with all chromosomes together) is encountered by most users, and this situation is the simplest to carry out.

              ./bin/PopLDdecay -InVCF  ALLchr.vcf.gz  -OutStat  LDDecay.stat.gz
              perl bin/Plot_OnePop.pl  -inFile  LDDecay.stat.gz -output  Out.Prefix

    Note:
      This will generate the two finale figures named 'Out.Prefix.png' and 'Out.Prefix.pdf'

2). Muti population
    This is common situation in the LD decay analysis. For example, if there are 50 samples (wild1, wild2, wild3...wild25, cul1, cul2, cul3...cul25) in the VCF file,
    To compare the LD decay of these two groups (wild vs cultivation), first of all, put their sample names into own file list for each group, column or row is ok.

             ./bin/PopLDdecay -InVCF  In.vcf.gz  -OutStat  wild.stat.gz  -SubPop wildName.list
             ./bin/PopLDdecay -InVCF  In.vcf.gz  -OutStat   cul.stat.gz  -SubPop culName.list
             #   created manually  muti.list by yourself
             perl bin/Plot_MutiPop.pl -inList  muti.list  -output  OutputPrefix

    Note:
      A. The <wildName.list> can list as follow(column or row is ok):
                      wild1
                      wild2
                      ...
                      wild25
      B. The format of <muti.list> had two columns, the file path of population result and the population flag, such as:
                      /ifshk7/BC_PS/Lddecay/wild.stat.gz   wild
                      /ifshk7/BC_PS/Lddecay/cul.stat.gz    cultivation

3). One population with multi-chr
    One population with multiple chromosome VCF files. For example, if there are 3 chromosomes VCF files (Chr1, Chr2 and Chr3) as the input.

            ./bin/PopLDdecay -InVCF  Chr1.vcf.gz  -OutStat  Chr1.stat.gz
            ./bin/PopLDdecay -InVCF  Chr2.vcf.gz  -OutStat  Chr2.stat.gz
            ./bin/PopLDdecay -InVCF  Chr3.vcf.gz  -OutStat  Chr3.stat.gz
            ls  `pwd`/Chr*.stat.gz   > chr.list
            perl bin/Plot_OnePop.pl -inList  chr.list  -output  OutputPrefix

   Note:
     A. It can run in parallel when calculating the chromosomes' statistics files.
     B. The files list only store the file path, which is diff with the multi-population list
     C. It will generate the file 'OutputPrefix.bin' is the summary statistics file of all chromosomes, and same format with the chromosomes' statistics files.
     D. The <chr.list> format can be generated by as above command 'ls Chr*.stat.gz   > chr.list'

4). Muti population with multi-chr
    Muti population with multiple chromosome VCF files. For example, if there are 2 chromosomes VCF files (Chr1, Chr2) as the input.

             ./bin/PopLDdecay -InVCF  Chr1.vcf.gz  -OutStat  W.Chr1.stat.gz -SubPop wildName.list
             ./bin/PopLDdecay -InVCF  Chr2.vcf.gz  -OutStat  W.Chr2.stat.gz -SubPop wildName.list
             ./bin/PopLDdecay -InVCF  Chr1.vcf.gz  -OutStat  C.Chr1.stat.gz -SubPop culName.list
             ./bin/PopLDdecay -InVCF  Chr2.vcf.gz  -OutStat  C.Chr2.stat.gz -SubPop culName.list
             ls  `pwd`/W.Chr*.stat.gz   > W.chr.list
             perl bin/Plot_OnePop.pl -inList  W.chr.list  -output  Wild.cat
             ls  `pwd`/C.Chr*.stat.gz   > C.chr.list
             perl bin/Plot_OnePop.pl -inList  C.chr.list  -output  Cul.cat
             perl bin/Plot_MutiPop.pl -inList  muti.list  -output  OutputPrefix

    Note:
     A. The format of <muti.list> had two columns , the file path of population result and the population flag, such as:
                      /ifshk7/BC_PS/Lddecay/Wild.cat.bin    wild
                      /ifshk7/BC_PS/Lddecay/Cul.cat.bin     cultivation

Reply:
@adamjstewart any thoughts?

Reply:
> This step here is to align the install path inline with its documentation

Based on the documentation, it sounds like they expect users to run the executable from the build directory instead of adding it to the `PATH`. So `./bin/PopLDdecay` would be right if it's not in your `PATH`, but `PopLDdecay` would be right if it is in the `PATH`. I would install it to `prefix.bin` like other packages.

Reply:
SGTM. Done. Thx @adamjstewart